### ISSUE TITLE
Few questions

### ISSUE THREAD

**Issue by comor** [KgBC/just-dice-bot#9]
1) Is there any option which after for example 5 loses will change bet to start value, like this:
1. 0.01 lose
2. 0.02 lose
3. 0.04 lose 
4. 0.08 lose
5. 0.16 lose 
---- start from the beginning
6. 0.01 

2) Can you explain how self.lose_rounds exacly works? Changing this value gives very strange results.

**Comment by KgBC** [comment]
Hi, thanks for your questions.
I'm currently testing a new set of settings, I'll do a expanatory site then. 
Just short info for those two questions:

1) currently not. but possible. I will not use it anyway, new settings include a safe balance setting which is not used for bet size calculations

2) lose_rounds is the number of rounds, you want to be able to lose. Bet size is proportional lower, that's why it looks strange. 
Currently Rounds calculation works correctly for 50:50 bets. Next version of calculation will correctly handle non-50-50 bets. (Well, not calculating, but simulating then, no worry its fast)

New Version will be up in some hours, I'll pass a link to the details I mentioned here :)

**Comment by comor** [comment]
Hi,
thanks for quick answers. This project is awesome.

Would like to report a bug. After few hours of running bot is crashing. Tested on Win 8, Win Xp. python v2.7

Traceback (most recent call last):
  File "just-dice-bot.py", line 195, in <module>
    JustDiceBet()
  File "just-dice-bot.py", line 69, in **init**
    saldo = self.do_bet(chance=self.chance, bet=bet)
  File "just-dice-bot.py", line 184, in do_bet
    raise Exception('bet was unsuccessful')
Exception: bet was unsuccessful

**Comment by KgBC** [comment]
Yea thanks, think I solved that already :)
There is huge documentation in the new version, see README and config-DEFAULT.py
Take a look and feel free to ask your questions.

I also have a new deal: I keep you win money, you tip 1% of your wins using auto-tip. It's a fair deal I think.
Sure you may also turn it off, but keeping it on makes sure the project is rolling. 

p.s.: just made a plus of 0.01689299 (to my new balance of 0.20233511) in 2 hours, 46 minutes. Dont trust me? Try the second (commented out) section in config yourself.  
THATS A +72% CALCULATED FOR 24 HOURS _woot_

**Comment by KgBC** [comment]
Reopening to note a feature request:
comor you wanted the possibility to re-start from the beginning:

I'll implement the following soon:

``````
"multiplier" : [ 2.0, 2.0, 2.0, 2.0, 'lose' ]  #multiply bet by 2.0 every round```

So four times multiply by 2.0, then 'lose' this series. which will result in what you want:

1. 0.01 lose --> x2
2. 0.02 lose --> x2
3. 0.04 lose --> x2
4. 0.08 lose --> x2
5. 0.16 lose --> 'lose'
---- start from the beginning
6. 0.01 

correct?
``````

**Comment by comor** [comment]
Yes, that's correct. Thanks.

**Comment by KgBC** [comment]
Cool, will be in next release. Hope you like the new features :)

**Comment by comor** [comment]
Will test them asap and hopefully donate :)

**Comment by KgBC** [comment]
Hm, thinking again, it's not so easy.

start bet is calculated from multiplyers and the rounds, we want to be able to lose.
So if we do:

```
"multiplier" : [ 2.0, 2.0, 2.0, 2.0, 'lose' ]
"lose_rounds": 5
```

This would match, so we e.g. bet 1 - 2 - 4 - 8 - 16 - lose (for a balance of 31)

Issue with this is, to what should the starting bet be calculated? Hint: if we calculate as normal we used all money on the 16 bet. so what to start with?

This doesn't make sence, we need a definition. As it was your request, I ask you :)

**Comment by comor** [comment]
You can let user to decide start bet value.

**Comment by KgBC** [comment]
This is a bit against the principles the bot follows, let me think about how it should be implemented.

Basic idea is: have the bot run forever. It does all (increasing bets, ...) on it own.

**Comment by KgBC** [comment]
Ok, I've got an idea ... tell me if you like it:

```
"multiplier" : [ 2.0, 2.0, 2.0, 2.0, 'lose10' ]
"lose_rounds": 5
```
- The lose10 means: we use 10% of our balance to calculate the first bet.
- So, for the same example as before: assume we own 310 btc (yea I know, just for easy calculation) :)
- 10% of 310 btc = 31 btc in this martingale streak.
- So bets are: 1 - 2 - 4 - 8 - 16 - then we accept, we lost and recalculate (from the new base 310-31 = 279 btc)
- usually lose_rounds and the rounds before 'loseX' should match here. To avoid config errors I would print an error if they don't match.

I did my calculations, I'm still unsure if this is a losing or winning strategy, but I would like to see someone trying it :)
